packages/devtools_app/lib/src/shared/ui/filter.dart

```diff
@@ -463,3 +465,156 @@ extension PatternListExtension on List<Pattern> {
         : cast<String>();
   }
 }
+
+class FilterField<T> extends StatefulWidget {
+  const FilterField({
+    super.key,
+    required this.controller,
+  });
+
+  final FilterControllerMixin<T> controller;
+
+  @override
+  State<FilterField<T>> createState() => _FilterFieldState<T>();
+}
+
+class _FilterFieldState<T> extends State<FilterField<T>> with AutoDisposeMixin {
+  late final TextEditingController queryTextFieldController;
+  late bool useRegExp;
+
+  @override
+  void initState() {
+    super.initState();
+    queryTextFieldController = TextEditingController(
+      text: widget.controller.activeFilter.value.queryFilter.query,
+    );
+    useRegExp = widget.controller.useRegExp.value;
+    addAutoDisposeListener(widget.controller.useRegExp, () {
+      useRegExp = widget.controller.useRegExp.value;
+    });
+  }
+
+  @override
+  void dispose() {
+    queryTextFieldController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
this seems to be duplicating a bunch of existing code. Can we modify our existing Filter functionality to fit this use case instead of creating two very similar widgets?

Reply:
This is possible but could this be left as an improvement to do in the future? In the name of moving fast to get the logging v2 change through I think that touching things outside of the logging_v2 directory gets into much slower and more rigorous territory.